### DISCUSSION TITLE
How to setup unquoted strings

### DISCUSSION THREAD

**Discussion by stevenbarragan** [tree-sitter/tree-sitter#1396] [Q&A] [answered]
I'd like to be able to use an unquoted string as a string e.g.

```rb
a_var = this_is_a_string 
b_var = a_var
```

I have this grammar
```js
module.exports = grammar({
  name: 'test',

  rules: {
    source_file: $ => repeat($.assign),
    assign:      $ => seq($.variable, "=", $._value),
    _value:      $ => choice($.variable, $.string),
    variable:    $ => /[a-z]+\w*/,
    string:      $ => prec.dynamic(1, choice(/\"\w*\"/, alias($.variable, 'string'))),
  },

  conflicts: $ => [
    [$._value, $.string]
  ]
});
```
Current results
```diff
  (assign
    (variable)
    (string)
  )
  (assign
-  (string)
+  (variable)
  )
```

The problem I'm having is that in the second assignation `a_var` always comes as a `(string)` not as a `(variable)`. Is there a way to make sure `a_var` usage is always a "variable" (after definition)?

**Comment by ahlinc** [comment, marked as answer]
Try to look on #1287 discussion. If you have some constrains for unquoted strings, this may help for parser to distinguish them. Also if your variables should be defined before their references then an external scanner with tracking of declared variables may help.

**Reply by stevenbarragan** [reply]
it seems that the best way would be to keep track of the declared variables with an external scanner then. thanks @ahlinc .
